alvr/dashboard/src/dashboard/components/setup_wizard.rs

```diff
@@ -161,13 +161,19 @@ This requires administrator rights!",
             ui.add_space(30.0);
             ui.horizontal(|ui| {
                 ui.add_space(15.0);
-                if ui
-                    .add_visible(self.page != Page::Finished, Button::new("Next"))
-                    .clicked()
-                {
-                    self.page = index_to_page(self.page as usize + 1);
+                let mut next_txt = "Next";
+                if self.page == Page::Finished {
+                    next_txt = "Finish";
+                }
+                let next_btn = ui.add(Button::new(next_txt));
+                if next_btn.clicked() {
                     if self.page == Page::Finished {
                         self.finished = true;
+                        response = Some(SetupWizardRequest::Close {
+                            finished: self.finished,
+                        });
+                    } else {
+                        self.page = index_to_page(self.page as usize + 1);
```

Comment:
Not sure if I understand the intent, why would changing the text create two separate buttons?

Reply:
You first put the text in a variable ("Next" or "Finished") depending on self.page, then create the button, then select the clicked() effect depending of self.page. What you could so instead is making a if-else block selecting the page, when make two separate buttons in the if and else branches.

Reply:
Is this more performant, or is it just better organisation? Thanks for the help!

Reply:
One final thing, should it be `ui.button("...")` instead of `ui.add(Button::new("..."))`? I presume this is the case, as this is used more often throughout the code

Reply:
> Is this more performant, or is it just better organisation?

egui is an immediate-mode UI library. This means that UI controls don't get instantiated or destroyed, but rendered frame by frame. It's much easier to code, the performance is worse but it doesn't matter, the UI is not a performance critical component.